src/code42cli/output_formats.py

```diff
@@ -81,41 +80,51 @@ def _requires_list_output(self):
 
 class DataFrameOutputFormatter:
     def __init__(self, output_format):
-        output_format = output_format.upper() if output_format else OutputFormat.TABLE
-        self.output_format = output_format
-        self._format_func = DataFrame.to_string
-        self._output_args = {"index": False}
+        self.output_format = (
+            output_format.upper() if output_format else OutputFormat.TABLE
+        )
 
-        if output_format == OutputFormat.CSV:
-            self._format_func = DataFrame.to_csv
-        elif output_format == OutputFormat.RAW:
-            self._format_func = DataFrame.to_json
-            self._output_args.update(
-                {
-                    "orient": "records",
-                    "lines": False,
-                    "index": True,
-                    "default_handler": str,
-                }
-            )
-        elif output_format == OutputFormat.JSON:
-            self._format_func = DataFrame.to_json
-            self._output_args.update(
-                {
-                    "orient": "records",
-                    "lines": True,
-                    "index": True,
-                    "default_handler": str,
-                }
-            )
+    def get_formatted_output(self, df, **kwargs):
+        if self.output_format == OutputFormat.JSON:
+            defaults = {
+                "orient": "records",
+                "lines": True,
+                "index": True,
+                "default_handler": str,
+            }
+            defaults.update(kwargs)
+            return df.to_json(**defaults)
+
+        elif self.output_format == OutputFormat.RAW:
+            defaults = {
+                "orient": "records",
+                "lines": False,
+                "index": True,
+                "default_handler": str,
+            }
+            defaults.update(kwargs)
+            return df.to_json(**defaults)
+
+        elif self.output_format == OutputFormat.CSV:
+            defaults = {"index": False}
+            defaults.update(kwargs)
+            df = df.fillna("")
+            return df.to_csv(**defaults)
+
+        elif self.output_format == OutputFormat.TABLE:
+            defaults = {"index": False}
+            defaults.update(kwargs)
+            df = df.fillna("")
+            return df.to_string(**defaults)
 
-    def _format_output(self, output, *args, **kwargs):
-        self._output_args.update(kwargs)
-        return self._format_func(output, *args, **self._output_args)
+        else:
+            raise Exception(
```

Comment:
Where does this get raised? Should this be a ` Code42CLIError` or a `click.Exception` ?

Reply:
At the very least this should have a more specific exception type (doesn't have to be a custom exception necessarily) to make catching this specific error easier if we want

Reply:
I did initially make it a Code42CLIError, but that caused a circular import. This should only get raised if someone makes a programming error somewhere, since the format options the user passes in _should_ be bounded by a `click.Choice` type. So we'd hopefully catch it in development/testing. 

Previously we were not raising anything if an unknown value was passed as the format, and just defaulting to Table. I figured it would be better to raise to make it clear where the problem was than have everything still function but maybe not do what was expected, for our own debugging/sanity.

The ExceptionHandlingGroup does catch all exceptions raised and if any are not known it just prints "unknown error, see logs" and logs the full stack trace. If we use a ClickException the message will get printed, but no other details logged.